tests/storage-spec.js

```diff
@@ -0,0 +1,179 @@
+                expect(this.languageStorage.getLanguageChoice('en-gb')).toBe(expectedResult);
+                expect(this.languageStorage.getLanguageChoice('en-GB')).toBe(expectedResult);
+            });
+
+            it('should have a default of the current selected language', function () {
+                expect(this.languageStorage.getLanguageChoice()).not.toBe(this.languageChoices[1]);
+                this.languageStorage.set('pl');
+                expect(this.languageStorage.getLanguageChoice()).toBe(this.languageChoices[1]);
+            });
+
+        });
+
+        describe('get method', function () {
+
+            it('should return the current selected language', function () {
+                expect(this.languageStorage.get()).not.toBe('se');
+                this.languageStorage.set('se');
+                expect(this.languageStorage.get()).toBe('se');
+            });
+
+        });
+
+        describe('set method', function () {
+
+            it('should set the cookie', function () {
+                expect(this.$cookies.get(this.languageStorage.getCookieSingature())).not.toBe('de');
+                this.languageStorage.set('de');
+                expect(this.$cookies.get(this.languageStorage.getCookieSingature())).toBe('de');
+            });
+
+            it('should emit an event', function () {
+                expect(this.$rootScope.$broadcast).not.toHaveBeenCalled();
+                this.languageStorage.set('es');
+                expect(this.$rootScope.$broadcast).toHaveBeenCalledWith(this.languageStorage.getEventSignature(), 'es');
+            });
+
+        });
+
+    });
+
+    describe('initially', function () {
+
+        beforeEach(function () {
+            document.cookie = 'selectedLanguage=pl';
+
+            this.languageChoices = [
+                {
+                    id: 'en_gb',
+                    label: 'English',
+                },
+                {
+                    id: 'pl',
+                    label: 'Polski',
+                },
+                {
+                    id: 'se',
+                    label: 'Svensk',
+                },
+            ];
+
+            this.setupModule();
+        });
+
+        it('should set the default language from the cookie', function () {
+            expect(this.languageStorage.get()).toBe('pl');
+        });
+
+    });
+
+    describe('when there is no cookie', function () {
+
+        beforeEach(function () {
+            document.cookie = 'selectedLanguage=;expires=Thu, 01 Jan 1970 00:00:01 GMT;';
+
+            this.languageChoices = [
+                {
+                    id: 'en_us',
+                    label: 'English',
+                },
+                {
+                    id: 'pl',
+                    label: 'Polski',
+                },
+            ];
+            this.setupModule();
+        });
+
+        it('should use the browser language', function () {
+            expect(this.languageStorage.get()).toBe('en_us');
```

Comment:
No, this test specifically tests the browser language, the next one tests the situation where the cookie doesn't match the language ie. it's the same case where there is no cookie at all. So this test has to stay like this or be removed enteirely